client/jetpack-cloud/sections/partner-portal/hooks/use-assign-licenses-to-site.ts

```diff
@@ -51,32 +51,50 @@ const useAssignLicensesToSite = (
 						( p ) => p.slug.substring( 0, 38 ) === productSlug
 					);
 
+					if ( ! selectedProduct ) {
+						return null;
+					}
+
 					return {
 						key,
 						product: selectedProduct,
 					};
 				} )
-				// If we can't determine which product a license is meant for, filter it out
-				.filter( ( { product } ) => Boolean( product ) )
+				// If we can't determine which product a license is meant for, filter it out.
+				// The Exclude<> usage here is to let Typescript know that the null case has been eliminated.
+				.filter( ( item ): item is Exclude< typeof item, null > => item !== null )
+				// Sort ascending by product id as a quick and dirty way to avoid race conditions between add-on
+				// products and their dependencies. For example, Jetpack Backup Add-on Storage requires that Jetpack
+				// Backup is present on the site, so we need to assign the Jetpack Backup license first.
+				// We rely on the product id because add-ons should practically always have a higher product id
+				// than the product they are an add-on for.
+				.sort( ( a, b ) => b.product.product_id - a.product.product_id )
```

Comment:
It seems "quick and dirty", but I consider it to be just a good, simple and practical implementation.